TheWarWithin/Priorities/ShamanRestoration.simc

```diff
@@ -1,13 +1,15 @@
 actions.precombat+=/earthliving_weapon
-actions.precombat+=/water_shield,if=buff.water_shield.up+buff.earth_shield.up+buff.lightning_shield.up<1+talent.elemental_orbit.rank
-actions.precombat+=/lightning_shield,if=buff.water_shield.up+buff.earth_shield.up+buff.lightning_shield.up<1+talent.elemental_orbit.rank
-actions.precombat+=/earth_shield,if=buff.water_shield.up+buff.earth_shield.up+buff.lightning_shield.up<1+talent.elemental_orbit.rank
+actions.precombat+=/tidecallers_guard
+actions.precombat+=/water_shield,if=buff.water_shield.up+buff.earth_shield.up+buff.lightning_shield.up<1+talent.elemental_orbit.rank|(!buff.water_shield.up&settings.healing_mode)
+actions.precombat+=/lightning_shield,if=(buff.water_shield.up+buff.earth_shield.up+buff.lightning_shield.up<1+talent.elemental_orbit.rank)&settings.second_shield=lightning_shield
+actions.precombat+=/earth_shield,if=(buff.water_shield.up+buff.earth_shield.up+buff.lightning_shield.up<1+talent.elemental_orbit.rank)&settings.second_shield=earth_shield
 actions.precombat+=/earth_elemental
 actions.precombat+=/potion
 
 actions+=/spiritwalkers_grace,moving=1,if=movement.distance>6
 # Interrupt of casts.
 actions+=/wind_shear
+actions+=/run_action_list,name=healing,if=settings.healing_mode
```

Comment:
Needs to be `call_action_list` if you want it to fall back to DPS abilities if no healing abilities are recommended.

Can add `,strict=1` to expedite addon decision-making (will fully skip when `healing_mode` is disabled instead of testing fork criteria for each entry in the list).